seed/seedwriter/writer.go

```diff
@@ -380,12 +422,34 @@ func (w *Writer) warningf(format string, a ...interface{}) {
 	w.warnings = append(w.warnings, fmt.Sprintf(format, a...))
 }
 
+func validateComponent(optComp *OptionsComponent) error {
+	if optComp.Name != "" {
+		if optComp.Path != "" {
+			return fmt.Errorf("cannot specify both name and path for component %q",
+				optComp.Name)
+		}
+		if err := snap.ValidateName(optComp.Name); err != nil {
+			return err
+		}
+	} else {
+		if !strings.HasSuffix(optComp.Path, ".comp") {
+			return fmt.Errorf("local option component %q does not end in .comp", optComp.Path)
+		}
+		if !osutil.FileExists(optComp.Path) {
+			return fmt.Errorf("local option component %q does not exist", optComp.Path)
+		}
+	}
+	return nil
+}
+
 // SetOptionsSnaps accepts options-referred snaps represented as OptionsSnap.
-func (w *Writer) SetOptionsSnaps(optSnaps []*OptionsSnap) error {
+func (w *Writer) SetOptionsSnaps(optSnaps []*OptionsSnap, pathToLocalComp map[string]*snap.ComponentInfo) error {
```

Comment:
thanks for this change, lets look at this again when we consider store components and see whether we should pass a []*OptionsComponent here in, or yet do something else